subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -480,9 +492,51 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t remainder, uint16_t laz
 	static struct mayfly mfy = {0, 0, &link, NULL, lll_scan_prepare};
 	static struct lll_prepare_param p;
 	struct ll_scan_set *scan = param;
+	struct lll_scan *lll;
 	uint32_t ret;
 	uint8_t ref;
 
+	lll = (struct lll_scan *)&scan->lll;
+
+	if (lll->duration) {
+		uint32_t num_int_to_skip;
+
+		/* -1 is a special value to indicate a prolonged interval */
+		if (lll->dur_counter == (uint32_t)-1) {
+			lll->dur_counter = 0;
+
+			ret = ticker_update(TICKER_INSTANCE_ID_CTLR,
+					TICKER_USER_ID_THREAD,
+					(TICKER_ID_SCAN_BASE +
+						ull_scan_handle_get(scan)),
+					0, 0, 0, 0,
+					1,
+					1,
+					NULL, NULL);
+			LL_ASSERT((ret == TICKER_STATUS_SUCCESS) ||
+				  (ret == TICKER_STATUS_BUSY));
+		} else {
+			lll->dur_counter += lll->interval;
```

Comment:
Use countdown, compared to addition operation.

Reply:
Also, need to take `lazy` value into account.

Reply:
Fixed